schema_generator/validation.py

```diff
@@ -124,7 +124,10 @@ def load_questions(schema_type, framework_slug, lot_slug):
     )
 
     manifest = loader.get_manifest(framework_slug, MANIFESTS[schema_type]['manifest']).filter(
-        {'lot': lot_slug},
+        {
+            "lot": lot_slug,
+            "item": "item",  # some dynamic list questions use `{{item}}` in template fields
+        },
```

Comment:
I'm not very familiar with content loader, so I'm having trouble understanding this. Is `DynamicList._make_dynamic_question#L453` (called from `DynamicList.filter`) not already adding `item` to the context?

Reply:
It does, but it is only called if `DynamicList.dynamic_field` is in the existing context (see https://github.com/alphagov/digitalmarketplace-content-loader/blob/master/dmcontent/questions.py#L335).

Reply:
But actually none of that code is reached anyway, because `dynamic=False` in the call to filter here.

Reply:
Ha - very true. So where will this `"item"` show up? I assume this won't be displayed to the user or appear in the API/logs?

Reply:
It never makes it into the schema, the error message displayed is treated as orthogonal to the validation check.

To be specific, in the current system, the API uses the schema to check a form response; if the validation fails the API generates a HTTP response containing the name of the validation check that failed; then the frontend looks at the response, cross-checks the name of the validation check with the content loader question, and looks up the error message for that validation failure.

Reply:
I think I now understand this. It looks as if validator only cares about the `name` field. And `item` only shows up in the `message` field.

So it's fine to give a placeholder because the validator doesn't use the field that this placeholder will appear in?

Might be nice to make this less confusing. But I don't know how one would do that without a fairly hefty amount of refactoring we probably don't have time for now. 

Could we perhaps be explicit that the only fields that use `item` are fields that aren't actually used by the validator?

Reply:
Assuming future changes hew to that design I do not expect the error messages to ever be present in the schema.

Reply:
> Could we perhaps be explicit that the only fields that use item are fields that aren't actually used by the validator?

Maybe I can write a test that asserts this...

Reply:
That's a good idea. I've made a suggestion on how we could change the wording to make it clearer for developers who don't already know how all the schema generation works.